include/RAJA/pattern/kernel/Conditional.hpp

```diff
@@ -69,6 +69,27 @@ struct Param : public internal::ParamBase {
   }
 };
 
+/*!
+ * An expression that returns the value of the specified RAJA::kernel
+ * segment.
+ *
+ * This allows run-time values to affect the control logic within
+ * RAJA::kernel execution policies.
+ */
+template <camp::idx_t SegId>
```

Comment:
Seg (and Param) should get moved out of Conditional.hpp and into their own files...  Seg.hpp and Param.hpp

Reply:
Okay I can work on adding the Seglist, Paramlist methods in there too.

Reply:
Also, Offset should be easy to implement now that you have Seg implemented!

Reply:
Should the offset struct be in its own file? or should it be in Seg.hpp?